metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -596,6 +596,11 @@ public void checkBrokerEpoch(int brokerId, long brokerEpoch) {
         }
     }
 
+    public long brokerEpoch(int brokerId) {
+        BrokerRegistration registration = brokerRegistrations.get(brokerId);
+        return registration == null ? -1 : registration.epoch();
```

Comment:
Returning -1 when the broker is a bit risky in my opinion because we also use -1 in the request. It would be better to return null for instance.

Reply:
Change made.

Reply:
Refactored. This change is reverted.